Create updated guidance on how to create a new page (#4517)
DFE-Digital/get-into-teaching-app: docs/content.md

It can take weeks or even months for changes to have an impact, so check the above regularly.

## Creating a new page

 **Make sure you start by creating a new branch on master**

> To do this you will need to
> 	- always open Visual Studio Code from https://github.com/DFE-Digital/get-into-teaching-app and open from the green code button
> 	- once Visual Studio Code is open, click the search bar and select 'Create new branch'
> 	- type in the name of your new branch and press enter
> 	- if your branch has been created, it will appear as a name in the bottom left hand corner, next to the blue tab of the codespace name 

Reviewer: ```suggestion
To do this you will need to:

- always open Visual Studio Code from https://github.com/DFE-Digital/get-into-teaching-app and open from the green code button
- once Visual Studio Code is open, click the search bar and select 'Create new branch'
- type in the name of your new branch and press enter
- if your branch has been created, it will appear as a name in the bottom left hand corner, next to the blue tab of the codespace name 
```
This is not showing as bullet points on the front end - not sure why! Maybe this will fix?

 Navigate to the left hand side bar, where you want the new page to sit
 Right hand click and select 'New file'

When you name the file you will have to name it depending on whether the file in going to be written in html or markdown. If the file is markdown then put .md at the end of the file, e.g. when-to-apply.md

If the file is html, start the file name with an underscore and then put .html.erb at the end of the file, e.g. _categories.html.erb

There are different reasons why you might want a file to be a specific type which you can discuss with a developer or interaction designer. Generally it is more straight forward to write something in markdown so this is used for most content files. However, html can be styled more easily so it is possible the file has been written in this way to feature a specific component such as a styled quote. 

If you have created a markdown page - you will need to fill in the following details at the top of the markdown file. You will not need to fill in the navigation details if the page sits in a sub category, e.g. Life as a teacher.

> title: title will appear in the browser tab
> heading: this is the h1 for the page - only needed if you want the h1 and the title to be different. If not, just add the title
> description: must start with |- and then you can write the text here
> related_content: list your related content links here, formatted as internal links
> navigation:  this is to decide where the tile for this page will sit on the page - you will need to check what the other pages are listed as. They will be formatted like '30.20' - '30' will stand for the category page (in this case, how to apply) and the '20' will be the order it sits in on the tiles. These usually are assigned in increments of 5, just in case you later want to move on further up it can be easily moved by assigning an in between number.  
> navigation_title: This is the title that will appear in the tiles
> navigation_description: This is the description that will appear in the tiles
> keywords: use Semrush to put keywords here, do a bulletpointed list using dashes like below:
>   - adviser
>   - advisor
>   - ITT

Reviewer: ```suggestion
title: title will appear in the browser tab

heading: this is the h1 for the page - only needed if you want the h1 and the title to be different. If not, just add the title

description: must start with |- and then you can write the text here

related_content: list your related content links here, formatted as internal links

navigation:  this is to decide where the tile for this page will sit on the page - you will need to check what the other pages are listed as. They will be formatted like '30.20' - '30' will stand for the category page (in this case, how to apply) and the '20' will be the order it sits in on the tiles. These usually are assigned in increments of 5, just in case you later want to move on further up it can be easily moved by assigning an in between number.  

navigation_title: This is the title that will appear in the tiles

navigation_description: This is the description that will appear in the tiles

keywords: use Semrush to put keywords here, do a bulletpointed list using dashes like below:
  - adviser
  - advisor
  - ITT
```

## Creating a subject page

> To create a new subject page, go to the folder content/life-as-a-teacher/explore-subjects. Then copy one of the current folders inside, e.g. english.

Reviewer: ```suggestion
To create a new subject page, go to the folder content/life-as-a-teacher/explore-subjects. Then copy one of the current folders inside, e.g. English.
```

Right click the folder and rename it to the subject you want to create. 

Inside the new folder you have created you will have 1 file:
-	_article.html/erb

This is where the content for the page will sit. 

Next, copy and paste a previous subject markdown (.md) file and rename it with the subject (e.g., business.md)

#### File changes:

The main content of subject pages should be written in HTML format in the article file. The structure of the subject pages are very similar so you should be able to use the subject you have copied as a guide.

Change all of the information in the article file that is not in a partial to be relevant to your subject. 
Then, you can optionally change the quote partials to be more relevant to your subject. Finally, look up the bursary and scholarship entitlement to the subject you are teaching. Then, change the information to the correct partial. All of the other information on the page should use the same partials so do not need to be edited. 

> Then update the markdown file to edit the hero banner image, hero banner text (title and title_paragraph) and the page’s card component. You will also need to ensure the links to the html erb files are updated

Reviewer: ```suggestion
Then update the markdown file to edit the hero banner image, hero banner text (title and title_paragraph) and the page’s card component. You will also need to ensure the links to the html erb files are updated.
```

Subject.md

> 

Reviewer: The yaml example below doesn't look the same as the updated version? E.g. I don't think we need the navigation bit any more?

>   **Make sure you're in the new branch when you create the doc and not master as you can't make changes to the master doc**

Reviewer: I think this creating a new page content should be with and above the 'creating a subject page' bit, so it's all together?

3. Navigate to the left hand side bar, where you want the new page to sit
4. Right hand click and select 'New file'

When you name the file you will have to name it depending on whether the file in going to be written in html or markdown. If the file is markdown then put .md at the end of the file, e.g. when-to-apply.md

> If the file is html then put .html.erb at the end of the file, e.g. when-to-apply.erb.html

Reviewer: ```suggestion
If the file is html, start the file name with an underscore and then put .html.erb at the end of the file, e.g. _categories.html.erb
```

> There are different reasons why you might want a file to be a specific type which you can discuss with a developer. Generally it is more straight forward to write something in markdown so this is used for most content files. However, html can be styled more easily so it is possible the file has been written in this way to feature a specific component such as a styled quote. 

Reviewer: ```suggestion
There are different reasons why you might want a file to be a specific type which you can discuss with a developer or interaction designer. Generally it is more straight forward to write something in markdown so this is used for most content files. However, html can be styled more easily so it is possible the file has been written in this way to feature a specific component such as a styled quote. 
```

If you have created a markdown page - you will need to fill in the top details:

> **If you have created a page that will appear in one of the category pages ie. How to apply**

Reviewer: Think need to add in the below that the navigation details aren't needed for sections that use the sub-category pages - e.g. Life as a teacher
